fdbserver/TLogServer.actor.cpp

```diff
@@ -1142,8 +1142,18 @@ ACTOR Future<Void> tLogPopCore(TLogData* self, Tag inputTag, Version to, Referen
 	int8_t tagLocality = inputTag.locality;
 	if (isPseudoLocality(tagLocality)) {
 		if (logData->logSystem->get().isValid()) {
-			upTo = logData->logSystem->get()->popPseudoLocalityTag(inputTag, to);
-			tagLocality = tagLocalityLogRouter;
+			// if the configuration change from multi-region to single region mode, the delayed pop created during
+			// multi-region stage should be skipped. Same thing applies to the backup worker
+			if (isPseudoLocality(inputTag.locality) &&
+			    logData->logSystem->get()->hasPseudoLocality(inputTag.locality)) {
+				upTo = logData->logSystem->get()->popPseudoLocalityTag(inputTag, to);
+				tagLocality = tagLocalityLogRouter;
+			} else {
+				TraceEvent(SevWarn, "TLogPopNoPseudoLocality", self->dbgid)
```

Comment:
Can you add a line here
```
ASSERT_WE_THINK(tagLocality == tagLocalityLogRouterMapped);
```
I think `tagLocalityBackup` is still valid, so an assertion is to make sure.

Reply:
Is there a possibility backup is also closed during the delay period?

Reply:
If backup (worker) is disabled, then there won't be a pop triggering this code path. Otherwise, `tagLocalityBackup` should always be present to be popped.